hack/usb-iso.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+
+parse_args() {
+   WRITE_USB=${WRITE_USB:-true}
+
+   USAGE="Usage: $0 [options]
+Options:
+    -d USB_DEV     Install to the given usb device, alternatively
+                   specify env var USB_DEV
+    -i IGNITION    The URL or path to the given Ignition config,
+                   alternatively specify env var IGNITION_URL
+    -o INSTALLER   The URL or path to the installer image,
+                   alternatively specify env var INSTALLER_IMAGE
+    -m METAL_IMG   The URL or path to the Metal image, 
+                   alternatively specify env var METAL_IMAGE
+    -b BLOCK_DEV   The block device on the remote node,
+                   alternatively specify env var BLOCK_DEV,
+		   default to /dev/sda
+    -h             This.
+
+This tool builds a USB key that can be used to install remote nodes when pxe boot is not feasible.
+"
+    while getopts "d:i:o:m:b:h" OPTION
+    do
+        case $OPTION in
+            d) USB_DEV="$OPTARG" ;;
+            i) IGNITION_URL="$OPTARG" ;;
+	    o) INSTALLER_IMAGE="$OPTARG" ;;
+            m) METAL_IMAGE="$OPTARG" ;;
+            b) BLOCK_DEV="$OPTARG" ;;
+            h) echo "$USAGE"; exit;;
+            *) echo "$USAGE"; exit 1;;
+        esac
+    done
+
+    if [[ -z "${USB_DEV}" && ${WRITE_USB} == "true" ]]; then
+	echo "usb device must be specified unless WRITE_USB is set to false!"
+	echo "$USAGE"
+	exit 1
+    fi
+    # normalize device name
+    USB_DEV=$(echo ${USB_DEV} | sed -r 's|/dev/||')
+    USB_DEV=/dev/${USB_DEV}
+
+    if [[ -z "${IGNITION_URL}" ]]; then
+	echo "ignition url must be specified!"
```

Comment:
if we are just passing an URL there, and not downloading the file, this validation is not needed